google/cloud/pubsub/internal/tracing_message_batch.h

```diff
@@ -38,6 +38,20 @@ class TracingMessageBatch : public MessageBatch {
  public:
   explicit TracingMessageBatch(std::unique_ptr<MessageBatch> child)
       : child_(std::move(child)) {}
+  TracingMessageBatch(
```

Comment:
Are these constructors just motivated by the tests?  Can we write the tests so they "naturally" leave the class in the state we want?

Reply:
The only alternative I can think of is a giant test that calls .SaveMessage then .Flush, then tests .FlushCallback...

I'm not sure if that's better. 

Do you have any ideas about how I could structure the tests?

Reply:
I would have written the giant test (maybe not that giant by using for loops?).  I am less worried about internal-only APIs like this, but feels a bit odd to test a class by setting a state only reachable via a test-only constructor.